src/testing/assertionTemplate.ts

```diff
@@ -127,11 +144,38 @@ export function assertionTemplate(renderFunc: () => DNode | DNode[]) {
 		const node = guard(findOne(render, selector));
 		return node.properties[property];
 	};
+	assertionTemplateResult.getProperties = (selector: string, property: string) => {
+		const render = renderFunc();
+		const node = guard(findOne(render, selector));
+		return node.properties;
+	};
 	assertionTemplateResult.getChildren = (selector: string) => {
 		const render = renderFunc();
 		const node = guard(findOne(render, selector));
 		return node.children || [];
 	};
+	assertionTemplateResult.replace = (selector: string, node: DNode) => {
+		return assertionTemplate(() => {
+			const render = renderFunc();
+			const node = guard(findOne(render, selector));
+			const parent = (node as any).parent;
+			const children = [...parent.children];
+			children.splice(children.indexOf(node), 1);
+			parent.children = [node, ...children];
```

Comment:
Wouldn't this always put the replacement node at the beginning of the children? You should be able to pass the replacement with `.splice()`.

Reply:
thanks was being silly here from the copy pasta from remove